src/main/java/com/couchbase/lite/support/CouchbaseLiteHttpClientFactory.java

```diff
@@ -120,62 +96,41 @@ public void deleteCookie(String name) {
         // 2. filter list to strip out the one we want to delete
         // 3. clear cookie store
         // 4. re-add all cookies except the one we want to delete
-        if (cookieStore == null) {
+        if (cookieJar == null)
             return;
-        }
-        List<Cookie> cookies = cookieStore.getCookies();
+
+        List<Cookie> cookies = cookieJar.loadForRequest(null);
         List<Cookie> retainedCookies = new ArrayList<Cookie>();
         for (Cookie cookie : cookies) {
-            if (!cookie.getName().equals(name)) {
+            if (!cookie.name().equals(name))
                 retainedCookies.add(cookie);
-            }
-        }
-        cookieStore.clear();
-        for (Cookie retainedCookie : retainedCookies) {
-            cookieStore.addCookie(retainedCookie);
         }
+        cookieJar.clear();
+        cookieJar.saveFromResponse(null, retainedCookies);
```

Comment:
Current implementation of `CookieJar.saveFromResponse(HttpUrl, List<Cookie>)` ignores first `HttpUrl` parameter. I will fix this with [the ticket](https://github.com/couchbase/couchbase-lite-java-core/issues/694).